pyccel/codegen/printing/pycode.py

```diff
@@ -124,13 +124,15 @@ def _print_FunctionDef(self, expr):
         return code
 
     def _print_Return(self, expr):
-        code = ''
+        rhs_list = []
+        lhs_list = []
+        expr_return_vars = []
         if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
-        if expr.expr:
-            ret = ','.join([self._print(i) for i in expr.expr])
-            code += 'return {}'.format(ret)
-        return code
+            rhs_list = [i.rhs for i in expr.stmt.body]
+            lhs_list = [i.lhs for i in expr.stmt.body]
+        expr_return_vars = [a for a in expr.expr if not lhs_list or a not in lhs_list]
+
+        return 'return ' + ','.join(self._print(i) for i in expr_return_vars + rhs_list)
```

Comment:
I find the structure of these few lines of code surprisingly confusing. For example, you set `expr_return_vars = []`, and then you always overwrite it. I suggest the following:

```python
def _print_Return(self, expr):

    rhs_list = [i.rhs for i in expr.stmt.body] if expr.stmt else []
    lhs_list = [i.lhs for i in expr.stmt.body] if expr.stmt else []
    expr_return_vars = [a for a in expr.expr if a not in lhs_list]
    
    return 'return ' + ','.join(self._print(i) for i in expr_return_vars + rhs_list)